File: samples/CS/19-Localization/RibbonItems.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace WinForms.Ribbon
{
    partial class RibbonItems
    {
        private bool exitOn = false;
        private RibbonButton _buttonDropA;
        private RibbonButton _buttonDropB;
        private RibbonButton _buttonDropC;

        public void Init()
        {
            _buttonDropA = ButtonOne;
            _buttonDropB = ButtonTwo;
            _buttonDropC = ButtonThree;
            _buttonDropA.Click += new EventHandler<EventArgs>(_buttonDropA_ExecuteEvent);
            _buttonDropB.Click += new EventHandler<EventArgs>(_buttonDropB_ExecuteEvent);
        }

        unsafe void _buttonDropA_ExecuteEvent(object sender, EventArgs e)
        {
            // load bitmap from file
            Bitmap bitmap = GetResourceBitmap("Drop32.bmp");
            bitmap.MakeTransparent();

            // set large image property
            _buttonDropA.LargeImage = new UIImage(Ribbon, bitmap); //.ConvertToUIImage(bitmap);
        }

        unsafe void _buttonDropB_ExecuteEvent(object sender, EventArgs e)
        {
            List<int> supportedImageSizes = new List<int>() { 32, 48, 64 };

            StringBuilder bitmapFileName = new StringBuilder();

            int selectedImageSize;
            if (supportedImageSizes.Contains(SystemInformation.IconSize.Width))
            {
                selectedImageSize = SystemInformation.IconSize.Width;
            }
            else
            {
                selectedImageSize = 32;
            }

            exitOn = !exitOn;
            string exitStatus = exitOn ? "On" : "Off";

            var bitmap = GetResourceBitmap(string.Format("Exit{0}{1}.bmp", exitStatus, selectedImageSize));
            bitmap.MakeTransparent();

            _buttonDropB.LargeImage = new UIImage(Ribbon, bitmap); //.ConvertToUIImage(bitmap);
        }

        Bitmap GetResourceBitmap(string name)
        {
            string resourceName = string.Format("_19_Localization.Res.{0}", name);
            using (var stream = this.GetType().Assembly.GetManifestResourceStream(resourceName))
            {
                var bitmap = new Bitmap(stream);
                return bitmap;
            }
        }

        public void Load()
        {
        }

    }
}

File: samples/VB/19-Localization/RibbonItems.vb
Imports System
Imports System.Text
Imports System.IO

Namespace Global.WinForms.Ribbon

	Partial Class RibbonItems
		Private exitOn As Boolean = False

		Public Sub Init()
			AddHandler ButtonOne.Click, AddressOf _buttonDropA_ExecuteEvent
			AddHandler ButtonTwo.Click, AddressOf _buttonDropB_ExecuteEvent
		End Sub

		Private Sub _buttonDropA_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			' load bitmap from file
			Dim bitmap_Renamed As Bitmap = GetResourceBitmap("Drop32.bmp")
			'bitmap_Renamed.MakeTransparent()

			' set large image property
			ButtonOne.LargeImage = New UIImage(Ribbon, bitmap_Renamed)
		End Sub

		Private Sub _buttonDropB_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			Dim supportedImageSizes As New List(Of Integer)() From {32, 48, 64}

			Dim bitmapFileName As New StringBuilder()

			Dim selectedImageSize As Integer
			If supportedImageSizes.Contains(SystemInformation.IconSize.Width) Then
				selectedImageSize = SystemInformation.IconSize.Width
			Else
				selectedImageSize = 32
			End If

			exitOn = Not exitOn
			Dim exitStatus As String = If(exitOn, "On", "Off")

			Dim bitmap_Renamed = GetResourceBitmap(String.Format("Exit{0}{1}.bmp", exitStatus, selectedImageSize))
			'bitmap_Renamed.MakeTransparent()

			ButtonTwo.LargeImage = New UIImage(Ribbon, bitmap_Renamed)
		End Sub

		Private Function GetResourceBitmap(ByVal name As String) As Bitmap
			Dim resourceName As String = String.Format("_19_Localization.Res.{0}", name)
			Using stream = Me.GetType().Assembly.GetManifestResourceStream(resourceName)
				Dim bitmap_Renamed = New Bitmap(stream)
				Return bitmap_Renamed
			End Using
		End Function

	End Class
End Namespace
